src/components/SavedNews.jsx

```diff
@@ -1,23 +1,45 @@
+import React, { useState, useEffect } from 'react'; 
 import SavedNewsCardsList from './SavedNewsCardsList';
-import SavedNewsHeader from './SavedNewsHeader';  
+import SavedNewsHeader from './SavedNewsHeader';
+import { getArticles } from '../utils/api';
 
 const SavedNews = ({
     isLoggedIn,
     currentUser,
     handleLogout,
     keyword,
 }) => {
+    const [savedArticles, setSavedArticles] = useState([]);
+
+    useEffect(() => {
+        if (isLoggedIn) {
+          getArticles()
+            .then((articles) => {
+              setSavedArticles(articles);
+            })
+            .catch((err) => console.error('Failed to fetch saved articles:', err));
+        }
+      }, [isLoggedIn]);
+
+      const keywords = [
+        ...new Set(savedArticles.map((article) => article.keyword).filter(Boolean)),
+      ];
+
     return ( 
         <>
         <SavedNewsHeader 
         isLoggedIn={isLoggedIn}
         currentUser={currentUser}
         handleLogout={handleLogout}
+        articleCount={savedArticles.length}
+        keywords={keywords}
```

Comment:
It's important to **format** code correctly, use the **indentation hierarchy**, and split your code. To make sure your code is formatted correctly, we suggest using Prettier (https://prettier.io/). It can be integrated with any code editor or IDE. If `Prettier` doesn’t format the code, then there are **validation errors** which should be fixed first. **Please, correct it everywhere in the code**


```jsx

  return (
    <>
      <SavedNewsHeader
        isLoggedIn={isLoggedIn}
        currentUser={currentUser}
        handleLogout={handleLogout}
        articleCount={savedArticles.length}
        keywords={keywords}
      />
      <SavedNewsCardsList
        isLoggedIn={isLoggedIn}
        currentUser={currentUser}
        keyword={keyword}
        savedArticles={savedArticles} // Pass articles down to SavedNewsCardsList
        setSavedArticles={setSavedArticles}
      />
    </>
  );
```